### DISCUSSION TITLE
Multiple projects/builds within one directory

### DISCUSSION THREAD

**Discussion by ecuapac** [clangd/clangd#1487] [Q&A] [answered]
I work on a project managed with Cmake. The structure is like:

```
/proj
/proj/library
/proj/library/CMakeLists.txt
/proj/application
/proj/application/CMakeLists.txt
/proj/build
/proj/build/compile_commands.json
/proj/build_application
/proj/build_application/compile_commands.json
/proj/src
```

`/proj/build` creates a shared library of the project, `/proj/build_application` creates an application using the library.

The documentation of clangd mentions that `compile_commands.json` is searched for in `build/` directories. I am using Visual Studio Code as an IDE and I am unable to resolve symbols of the application source. I assume clangd is not looking up the `/proj/build_application` directory.

Is there a way to tell clangd to also search other directories than just `build/`?

**Comment by HighCommander4** [comment]
You can point clangd at a `compile_commands.json` in a custom location using the `--compile-commands-dir` command-line flag, but this causes clangd to **only** look in that directory, so it's suitable if you use different workspaces for your library and your application but not if you want to work on library + application code in the same workspace.

For the latter case, my best suggestion is to add a build step that runs a simple `jq` command to merge the entries from the two `compile_commands.json` files into a single `compile_commands.json` file in the workspace root.

**Comment by ecuapac** [comment, marked as answer]
My current solution is using a `.clangd` project configuration file. See [clangd configuration documentation](https://clangd.llvm.org/config).

The configuration file is in `/proj/.clangd` and is:
```
If:
  PathMatch: application/.*

CompileFlags:
  CompilationDatabase: build_application/
```
... meaning that, files matching the given path have a compilation database `build_application/compile_commands.json`.

**Reply by HighCommander4** [reply]
Oh, right! Forgot about that possibility, that does sound like the easiest approach in this case.
